File: src/main/java/com/tianli/auth/grasscutterauth/controller/LoginController.java
package com.tianli.auth.grasscutterauth.controller;

import com.tianli.auth.grasscutterauth.domain.User;
import com.tianli.auth.grasscutterauth.service.LoginService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;
import reactor.core.publisher.Mono;

@RestController
@RequiredArgsConstructor
@RequestMapping("/auth/login")
class LoginController {

    private final LoginService loginService;

    @GetMapping("/get/{id}")
    @ResponseBody
    public Mono<User> getUserById(@PathVariable Long id) {
        return Mono.just(loginService.getUserById(id));
    }

}

File: src/main/java/com/tianli/auth/grasscutterauth/controller/LoginController.kt
package com.tianli.auth.grasscutterauth.controller;

import com.tianli.auth.grasscutterauth.domain.User
import com.tianli.auth.grasscutterauth.service.LoginService
import freemarker.template.Configuration
import lombok.RequiredArgsConstructor
import org.springframework.mail.javamail.JavaMailSender
import org.springframework.mail.javamail.MimeMessageHelper
import org.springframework.web.bind.annotation.*
import reactor.core.publisher.Mono
import java.io.StringWriter

@RestController
@RequiredArgsConstructor
@RequestMapping("/auth/login")
class LoginController(private val loginService:LoginService) {

    @GetMapping("/get/{id}")
    fun getUserById(@PathVariable id:Long):Mono<User>  {
        return Mono.just(loginService.getUserById(id))
    }

}
